Intersect.Client/Entities/Entity.cs

```diff
@@ -1386,6 +1386,147 @@ public virtual void DrawName(Color textColor, Color borderColor = null, Color ba
+                        // Position the tag as prefix (2 pixels left from the name label).
+                        x = nameCentHorPos - (nameSize.X / 2) - tagTexture.GetWidth() - 6;
+                        y = nameVertPos + (nameSize.Y / 2) - (tagTexture.GetHeight() / 2);
+
+                        break;
+                    case NpcTagPos.Suffix:
+                        // Position the tag as suffix (2 pixels right from the name label).
+                        x = nameCentHorPos + (nameSize.X / 2) + 6;
+                        y = nameVertPos + (nameSize.Y / 2) - (tagTexture.GetHeight() / 2);
+
+                        break;
+                }
+                // And finally, we draw the tag.
+                if (tagTexture != null)
+                {
+                    Graphics.DrawGameTexture(tagTexture, x, y);
+                }
+            }
+            // Player Tags
+            if (Options.Player.ShowTags && this is Player && customPlayerTag)
+            {
+                // Custom Player Tag.
+                tagFileName = $@"Player_{entityName}.png";
+                // Now that we have the name of the sprite file, we load it as a texture.
+                GameTexture tagTexture = Globals.ContentManager.GetTexture(GameContentManager.TextureType.Tag, tagFileName);
+                // Before we draw the sprite, lets have it's position set.
+                switch (playerTagPos)
+                {
+                    case PlayerTagPos.Above:
+                    default:
+                        // Position the tag 2 pixels above the name label.
+                        x = nameCentHorPos - (tagTexture.GetWidth() / 2);
+                        y = nameVertPos - tagTexture.GetHeight() - 2;
+
+                        break;
+                    case PlayerTagPos.Under:
+                        // Position the tag 2 pixels under the name label.
+                        x = nameCentHorPos - (tagTexture.GetWidth() / 2);
+                        y = nameVertPos + nameSize.Y + 2;
+
+                        break;
+                    case PlayerTagPos.Prefix:
+                        // Position the tag as prefix (2 pixels left from the name label).
+                        x = nameCentHorPos - (nameSize.X / 2) - tagTexture.GetWidth() - 6;
+                        y = nameVertPos + (nameSize.Y / 2) - (tagTexture.GetHeight() / 2);
+
+                        break;
+                    case PlayerTagPos.Suffix:
+                        // Position the tag as suffix (2 pixels right from the name label).
+                        x = nameCentHorPos + (nameSize.X / 2) + 6;
+                        y = nameVertPos + (nameSize.Y / 2) - (tagTexture.GetHeight() / 2);
+
+                        break;
+                }
+                // And finally, we draw the tag.
+                if (tagTexture != null)
+                {
+                    Graphics.DrawGameTexture(tagTexture, x, y);
+                }
+            }
```

Comment:
This entire section should be relocated to an `override` function in `Player.cs`.

Reply:
👍🏽 

```
        public override void DrawTag()
        {
            // Variables
            string playerName = this.Name;
            bool customPlayerTag = Options.Player.CustomTagIcons.Contains(playerName);
            var nameSize = Graphics.Renderer.MeasureText(playerName, Graphics.EntityNameFont, 1);
            var nameCentHorPos = (int)Math.Ceiling(GetCenterPos().X);
            var nameVertPos = GetLabelLocation(LabelType.Name);
            var tagPos = Options.Player.TagPosition;
            float x, y;
            // Feature and Entity check
            if (!Options.Player.ShowTags || !(this is Player))
            {
                return;
            }
            // Player Tags
            else if (customPlayerTag)
            {
                // Lets load the player's custom Tag texture.
                var tagTexture = Globals.ContentManager.GetTexture(GameContentManager.TextureType.Tag, $@"Player_{playerName}.png");
                // If the texture is null, we do nothing.
                if (tagTexture == null)
                {
                    return;
                }
                // Before we draw the sprite, lets have it's position set.
                switch (tagPos)
                {
                    case TagPosition.Above:
                    default:
                        // Position the tag 2 pixels above the name label.
                        x = nameCentHorPos - (tagTexture.GetWidth() / 2);
                        y = nameVertPos - tagTexture.GetHeight() - 2;

                        break;
                    case TagPosition.Under:
                        // Position the tag 2 pixels under the name label.
                        x = nameCentHorPos - (tagTexture.GetWidth() / 2);
                        y = nameVertPos + nameSize.Y + 2;

                        break;
                    case TagPosition.Prefix:
                        // Position the tag as prefix (2 pixels left from the name label).
                        x = nameCentHorPos - (nameSize.X / 2) - tagTexture.GetWidth() - 6;
                        y = nameVertPos + (nameSize.Y / 2) - (tagTexture.GetHeight() / 2);

                        break;
                    case TagPosition.Suffix:
                        // Position the tag as suffix (2 pixels right from the name label).
                        x = nameCentHorPos + (nameSize.X / 2) + 6;
                        y = nameVertPos + (nameSize.Y / 2) - (tagTexture.GetHeight() / 2);

                        break;
                }
                // And finally, we draw the tag.
                if (tagTexture != null)
                {
                    Graphics.DrawGameTexture(tagTexture, x, y);
                }
            }
        }
```